frontend/javascripts/oxalis/model/sagas/annotation_saga.ts

```diff
@@ -159,4 +178,75 @@ export function* watchAnnotationAsync(): Saga<void> {
   );
   yield* takeLatest("EDIT_ANNOTATION_LAYER", pushAnnotationLayerUpdateAsync);
 }
-export default [warnAboutSegmentationZoom, watchAnnotationAsync];
+
+export function* acquireAnnotationMutexMaybe(): Saga<void> {
+  yield* take("WK_READY");
+  const allowUpdate = yield* select((state) => state.tracing.restrictions.allowUpdate);
+  const annotationId = yield* select((storeState) => storeState.tracing.annotationId);
+  if (!allowUpdate) {
+    return;
+  }
+  const othersMayEdit = yield* select((state) => state.tracing.othersMayEdit);
+  const ONE_MINUTE = 1000 * 60;
+  let isInitialRequest = true;
+  let doesHaveMutex = false;
+  let shallTryAcquireMutex = othersMayEdit;
+
+  function onMutexStateChanged(canEdit: boolean, blockedByUser: APIUserCompact | null | undefined) {
+    if (canEdit) {
+      Toast.close("MutexCouldNotBeAcquired");
+      if (!isInitialRequest) {
+        Toast.success(messages["annotation.acquiringMutexSucceeded"]);
+        location.reload();
+      }
+    } else {
+      const message =
+        blockedByUser != null
+          ? messages["annotation.acquiringMutexFailed"]({
+              userName: `${blockedByUser.firstName} ${blockedByUser.lastName}`,
+            })
+          : messages["annotation.acquiringMutexFailed.noUser"];
+      Toast.error(message, { sticky: true, key: "MutexCouldNotBeAcquired" });
+    }
+  }
+
+  function* tryAcquireMutex(): Saga<void> {
```

Comment:
i would rename it to something like `tryAcquireMutexContinuously` to signal that this is acquiring continuously.